nsenter/nsexec.c

```diff
@@ -113,6 +124,28 @@ void nsexec()
 	}
 
 	if (setjmp(env) == 1) {
+		if (setsid() == -1) {
+			pr_perror("setsid failed");
+			exit(1);
+		}
+		if (consolefd != -1) {
+			if (ioctl(consolefd, TIOCSCTTY, 0) == -1) {
```

Comment:
Ignore comment. I see why we do it this way. We need to get the fd before the clone, else the console path isn't accessible.